### PR TITLE
Move C API versioning into Version.xcconfig

### PR DESCRIPTION
**Pull request by dmbryson**
Implements rdar://problem/38434243
Fixes rdar://problem/38400756

### CODE REVIEW

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
"This time for sure!"

Fixes CMake Swift bindings build.
Fixes Xcode install headers build.

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
It might take more than one CI invocation after a rebase to get it to hear you

@swift-ci please smoke test

File: llbuild.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2628,12 +2685,31 @@
...
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
+		40513F1B2057626A00DCA088 /* Generate version.h */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 12;
+			files = (
+			);
+			inputPaths = (
+				"",
```
Comment (ddunbar) [inline llbuild.xcodeproj/project.pbxproj]: I think there is a stray empty path here...

Comment (dmbryson) [reply llbuild.xcodeproj/project.pbxproj]: huh, interesting

Comment (dmbryson) [reply llbuild.xcodeproj/project.pbxproj]: Indeed there was.

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
Well, I've double checked Xcode (llbuild and PerfTest schemes), cmake/make test and cmake/ninja test (both with Swift bindings enabled), and installhdrs.  🤞

**Pull request merged**